pandas/core/nanops.py

```diff
@@ -326,14 +329,19 @@ def nanmean(values, axis=None, skipna=True):
 @bottleneck_switch()
 def nanmedian(values, axis=None, skipna=True):
 
+    if not skipna and isnull(np.sum(values)):
+        return np.median(values, axis=axis)
+
     values, mask, dtype, dtype_max = _get_values(values, skipna)
```

Comment:
then define

```
if _np_under_version1p19:
       agg1d = get_median
       aggaxis = lambda values, axis: np.apply_along_axis(get_median, axis, values)
else:
      agg1d = np.nanmedian
      aggaxis = lambda values, axis: np.nanmedian(values, axis)
```

then the code is almost the same as before, just replace with ``agg1d`` and ``aggaxis``